### ISSUE TITLE
Question on loading pretrained weights

### ISSUE THREAD

**Issue by kcs6568** [open-mmlab/mmclassification#573]
### Describe the question you meet

Hello. I'm going to train AlexNet model now.

I assigned the pre-trained weight file provided by the official pythorch GitHub to the load_from variable in the config file.

Warning continues to be output whenever pre-trained weight is used.

Warning is as follows.:
```python
WARNING - The model and loaded state dict do not match exactly

unexpected key in source state_dict: features.0.weight, features.0.bias, features.3.weight, features.3.bias, features.6.weight, features.6.bias, features.8.weight, features
es.10.bias, classifier.1.weight, classifier.1.bias, classifier.4.weight, classifier.4.bias, classifier.6.weight, classifier.6.bias

missing keys in source state_dict: backbone.features.0.weight, backbone.features.0.bias, backbone.features.3.weight, backbone.features.3.bias, backbone.features.6.weight, backbone.features
eatures.8.weight, backbone.features.8.bias, backbone.features.10.weight, backbone.features.10.bias, backbone.classifier.1.weight, backbone.classifier.1.bias, backbone.classifier.4.w
.4.bias, backbone.classifier.6.weight, backbone.classifier.6.bias
```
Warning is that the weight value and bias value of Convolution Layer do not match, but I don't understand well.

I don't know if the value is not correct or the architecture is not correct.

Can I ignore this Warning?
If I ignore it, it doesn't affect the performance?

Currently, I'm going to use the CUB-200 dataset.
Please refer to the config file here.

```python
model = dict(
    type='ImageClassifier',
    backbone=dict(type='AlexNet', num_classes=200),
    neck=None,
    head=dict(
        type='ClsHead',
        loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
        topk=(1, 5),
    ),
    pretrained='/root/volume/weights/imagenet/alexnet-owt-4df8aa71.pth')
```

**Comment by mzr1996** [comment]
Please try this config:
```python
model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='AlexNet',
        num_classes=200,
        init_cfg=dict(type='Pretrained', checkpoint='/root/volume/weights/imagenet/alexnet-owt-4df8aa71.pth'),
    ),
    neck=None,
    head=dict(
        type='ClsHead',
        loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
        topk=(1, 5),
    ))
···

**Comment by kcs6568** [comment]
Thank you for your answer.

AlexNet Backbone does not have init_cfg argument.
The warning did not occur when init_cfg in the backbone was entered as an argument in the Image Classifier.
The modified cfg is followed.

```python
model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='AlexNet',
        num_classes=200,
    ),
    neck=None,
    head=dict(
        type='ClsHead',
        loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
        topk=(1, 5),
    ),
    init_cfg=dict(type='Pretrained', checkpoint='/root/volume/weights/imagenet/alexnet-owt-4df8aa71.pth'))
```

However, when loading pre-trained weight from a long time ago, the following log was always output by the logger.

```python
mmcv - INFO - Use load_from_local loader
```

However, the cfg you told me does not show logs and does not print warning.

So I'm not sure if the pretrained weight was loaded.

**Comment by mzr1996** [comment]
Well, please modify the `mmcls/models/backbones/alexnet.py` as below:
```python
@BACKBONES.register_module()
class AlexNet(BaseBackbone):

    def __init__(self, num_classes=-1, init_cfg=None):
        super(AlexNet, self).__init__(init_cfg=init_cfg)

```
And then use my config. This is a bug, and we will fix it later.

**Comment by kcs6568** [comment]
Thanks for your reply! I solved that problems! Thank you!
